### PR TITLE
chromium: add libglvnd to rpath

### LINKED ISSUE #268490: chromium/electron/element-desktop on wayland fails to find libEGL.so
### Describe the bug

Running `element-desktop` (from current unstable, 1.11.47) on Wayland doesn't open a window, but the notification icon is there and element-desktop does not crash.

### Steps To Reproduce
1. Run `element-desktop` from current nixos-unstable on Wayland

### Expected behavior
`element-desktop` should open.

### Additional context

Relevant parts of the log:
```
[21803:1119/123816.722060:ERROR:angle_platform_impl.cc(44)] Display.cpp:1066 (initialize): ANGLE Display::initialize error 12289: Could not dlopen native EGL: libEGL.so.1: cannot open shared object file: No such file or directory
ERR: Display.cpp:1066 (initialize): ANGLE Display::initialize error 12289: Could not dlopen native EGL: libEGL.so.1: cannot open shared object file: No such file or directory
[21803:1119/123816.722150:ERROR:gl_display.cc(520)] EGL Driver message (Critical) eglInitialize: Could not dlopen native EGL: libEGL.so.1: cannot open shared object file: No such file or directory
[21803:1119/123816.722184:ERROR:gl_display.cc(790)] eglInitialize OpenGL failed with error EGL_NOT_INITIALIZED, trying next display type
[21803:1119/123816.722273:ERROR:angle_platform_impl.cc(44)] Display.cpp:1066 (initialize): ANGLE Display::initialize error 12289: Could not dlopen native EGL: libEGL.so.1: cannot open shared object file: No such file or directory
ERR: Display.cpp:1066 (initialize): ANGLE Display::initialize error 12289: Could not dlopen native EGL: libEGL.so.1: cannot open shared object file: No such file or directory
[21803:1119/123816.722311:ERROR:gl_display.cc(520)] EGL Driver message (Critical) eglInitialize: Could not dlopen native EGL: libEGL.so.1: cannot open shared object file: No such file or directory
[21803:1119/123816.722339:ERROR:gl_display.cc(790)] eglInitialize OpenGLES failed with error EGL_NOT_INITIALIZED
[21803:1119/123816.722364:ERROR:gl_display.cc(824)] Initialization of all EGL display types failed.
[21803:1119/123816.722393:ERROR:gl_ozone_egl.cc(26)] GLDisplayEGL::Initialize failed.
```

This issue can be fixed by wrapping element-desktop with `--prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ pkgs.libglvnd ]}"`.

See also #243372 and #249853 for a similar issue happening with VSCode. The correct fix would probably be to unbundle `libglvnd` from `electron` itself and use our packaged `libglvnd` directly.

### Notify maintainers

@Ma27 @yu-re-ka 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.1-zen1, NixOS, 23.11 (Tapir), 23.11.20231117.c757e9b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(sophie): `""`
 - channels(root): `"nixos"`
 - nixpkgs: `/etc/channels/nixpkgs`
```

### PR DESCRIPTION
**Pull request by 999eagle**
## Description of changes

This should fix issues like
- https://github.com/NixOS/nixpkgs/pull/249853
- https://github.com/NixOS/nixpkgs/issues/268490
- https://discourse.nixos.org/t/electron-apps-cant-find-opengl/35713/14

for all chromium and all electron apps under Wayland.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

I've tested this by building `electron` and `element-desktop` with this. This change fixes #268490.

### CODE REVIEW

**Review by emilylange — APPROVED** [review]
thanks, looks good to me.

I'll look into adding wayland to our chromium VM test (though, to be fair, this wouldn't have caught it, as chromium opens just fine 🫠).

**Comment by 999eagle** [comment]
> though, to be fair, this wouldn't have caught it, as chromium opens just fine 🫠

Maybe there's a way to automatically check `chrome://gpu` in the test for the status of hardware acceleration? That most likely would've caught this.

**Comment by emilylange** [comment]
Looks like https://github.com/NixOS/nixpkgs/blob/e89c150f61958b693d3da1a1ff6845f59acc1b83/nixos/tests/chromium.nix#L251-L254 has some logic that could somewhat easily be extended.

I still fear that hardware acceleration in our qemu VMs behave differently, but we will see I guess /shrug

File: pkgs/applications/networking/browsers/chromium/common.nix
Comment (NickCao) [inline pkgs/applications/networking/browsers/chromium/common.nix:483]: Isn't libGL already libglvnd on linux?

Comment (999eagle) [reply pkgs/applications/networking/browsers/chromium/common.nix:483]: Yes, `libGL/lib` actually is just a symlink to `libglvnd/lib`. I really don't know why but this single change fixes electron-apps in Wayland for me.
Without this change, `electron` contains just `libGL` in its runpath and then the Wayland backend fails to find and load `libEGL.so.1`. Adding `libglvnd` (which is the exact same path `libGL` links to) to the runpath somehow fixes the issue and I have no idea why.
I guess some dynamic loader somewhere inside `electron` itself doesn't respect symlinks when searching the runpath for `libEGL.so.1`?

**Comment by yu-re-ka** [comment]
I don't think this fix actually works. I'm still getting the same symptoms (tray icon, but the gui doesn't open, and libEGL.so errors in the log) with nixpkgs fdc53c5b2f1ee4446dd5709b06a071353d8f991a

**Comment by emilylange** [comment]
Oh, that's a bummer.
I've been a bit trigger-happy with the merge button here, honestly.

Still haven't come around to write tests for this :(

**Comment by 999eagle** [comment]
Running element-desktop from my branch (that is, https://github.com/NixOS/nixpkgs/pull/268715/commits/c475ac3248877ce62bca18c659c7e9e2f9ee651e) and element-desktop from https://github.com/NixOS/nixpkgs/commit/fdc53c5b2f1ee4446dd5709b06a071353d8f991a uses a different electron. The latter one is already cached in hydra and the former I still have to build myself. https://github.com/NixOS/nixpkgs/commit/fdc53c5b2f1ee4446dd5709b06a071353d8f991a indeed still has the same issue for me as well....
I'll continue investigating in the next few days when I have more time.

**Comment by yu-re-ka** [comment]
@999eagle that's because your branch is based on a master from 4th Nov, before the last staging cycle was merged. So on your branch the issue was never introduced and everything works even without the fix.

**Comment by lilyinstarlight** [comment]
I suppose for people subscribed here, I have opened #269345 with what is hopefully a proper fix. Needs testing, but I'm pretty confident in having found the immediate cause and a reasonable fix

**Pull request merged**
